File: FileReadingCMDArguments/reading.py
import sys

if __name__ == '__main__':
    script_name = sys.argv[0]
    if(sys.argv.__len__()>=1):
        input_file_path = sys.argv[1]
    print(f"the script Name is :  {script_name}")
    print(f"The file is : {input_file_path}")   

    with open(input_file_path, 'r') as file:
        # for line in file:
        #     print(line, end='')     
        first_name = file.readline()
        second_name = file.readline()
        age = int(file.readline())
        college = (file.readline()).split()

        print(f"the first name is {first_name} second name is {second_name}, age is {age} ")
        print(f"College is {college[0]} {college[1]} {college[2]}")

File: FileReadingCMDArguments/Reading.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.util.Iterator;
import java.util.stream.Stream;



public class Reading {
    public static void main(String[] args) {
        String filePath = args[0];
        try {
            BufferedReader br = new BufferedReader(new FileReader(filePath));
            // String firstName = br.readLine();
            // System.out.println(firstName);
            // String secondName = br.readLine();
            // System.out.println(secondName);
            // int age = Integer.parseInt(br.readLine());
            // System.out.println(age);
            // String collegeName = br.readLine();
            // String [] college = collegeName.split(" ");
            // System.out.println(college[0] + " " + college[1] + " " + college[1]);

            Stream<String> s =  br.lines();
            // System.out.println(s.count()); // 6   (one Stream can be called one then it is automatically closed)
            Iterator<String> it = s.iterator();
            while(it.hasNext()){
                System.out.println(it.next());
            }

            

            // String line;
            // while((line = br.readLine())!=null){
            //     System.out.println("the line is "+ line);
            // }
 
        } catch (FileNotFoundException e) {
            System.out.println("file not found" + e.toString());
        }
        catch(Exception e){
            System.out.println("something went wrong" + e.toString());
        }
    }    
}
